glyphs-reader/src/font.rs

```diff
@@ -1702,31 +1658,136 @@ impl TryFrom<RawFont> for Font {
     }
 }
 
+fn preprocess_unparsed_plist(s: String) -> String {
+    // Glyphs has a wide variety of unicode definitions, not all of them parser friendly
+    // Make unicode always a string, without any wrapping () so we can parse as csv, radix based on format version
+    let unicode_re =
+        Regex::new(r"(?m)^(?P<prefix>\s*unicode\s*=\s*)[(]?(?P<value>[0-9a-zA-Z,]+)[)]?;\s*$")
+            .unwrap();
+    // Some special glyph names can be parsed as floats so we enquote them
+    let glyphname_re =
+        Regex::new(r"(?mi)^(?P<prefix>\s*glyphname\s*=\s*)(?P<value>(?:infinity|inf|nan));\s*$")
+            .unwrap();
+    let s = unicode_re.replace_all(&s, r#"$prefix"$value";"#);
+    let s = glyphname_re.replace_all(&s, r#"$prefix"$value";"#);
```

Comment:
if a glyph is named "infinity", "inf" or "nan" and the name is un-quoted (as Glyphs.app does when a string only contains ascii), our plist parser misunderstands it as a float
https://doc.rust-lang.org/std/primitive.f64.html#grammar
so I uses regex to add the quotes and make sure they stay as strings

We had code (that I removed here) which would fix up the parsed glyph plist dict and unconditionally replace "glyphname" typed float with a hard-coded "infinity" string (ignoring "inf" or "nan").

I think the python openstep-plist parser (used by glyphsLib and thus fontmake) does not have this problem because it can detect that an unquoted string is neither an integer nor a float (https://github.com/fonttools/openstep-plist/blob/c9acd408ba3a374ba41dd62f595f43fa2e5bfa6f/src/openstep_plist/parser.pyx#L267-L309) and thus treat it as string; I think our current rust plist parser attempts to parse as number and if that succeeds (as in the case of "infinity") it treats it as such, so we need to prevent that -- or change the parser (but I haven't even looked into that.. would you like me to?)

Reply:
I changed the plist parser itself to always treat "infinity", "inf" and "nan" as strings, so we don't need the regex hack any more. I don't think inf or NaN are useful in font source formats